src/schema.rs

```diff
@@ -64,3 +65,16 @@ impl Schema {
         todo!()
     }
 }
+
+/// Provides an Iterator which returns [Type]s inside a [Schema]
+pub struct SchemaTypeIterator<'a> {
```

Comment:
I have a couple of questions: 
- This looks like `SchemaTypeIterator` is just a wrapper around an `Iter`. Can you explain to me why we need to have a wrapper?
- Is it considered idiomatic to use non-descriptive lifetime identifiers like `'a`?

Reply:
We are just trying to hide the usage of `TypeStore` into a `SchemaTypeIterator` rather than returning an `Iter` and yes, we can use non-descriptive lifetime here which is used with `Iter` as well as `Type` reference. Does this answer your question?